url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/StandardBeanInfo.java

```diff
@@ -0,0 +1,1505 @@
+
+        if (properties == null) {
+            properties = introspectProperties(stopClass);
+        }
+
+        if (events == null) {
+            events = introspectEvents();
+        }
+    }
+
+    @Override
+    public BeanInfo[] getAdditionalBeanInfo() {
+        return null;
+    }
+
+    @Override
+    public EventSetDescriptor[] getEventSetDescriptors() {
+        return events;
+    }
+
+    @Override
+    public MethodDescriptor[] getMethodDescriptors() {
+        return methods;
+    }
+
+    @Override
+    public PropertyDescriptor[] getPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public BeanDescriptor getBeanDescriptor() {
+        if (beanDescriptor == null) {
+            if (explicitBeanInfo != null) {
+                beanDescriptor = explicitBeanInfo.getBeanDescriptor();
+            }
+            if (beanDescriptor == null) {
+                beanDescriptor = new BeanDescriptor(beanClass);
+            }
+        }
+        return beanDescriptor;
+    }
+
+    @Override
+    public int getDefaultEventIndex() {
+        return this.defaultEventIndex;
+    }
+
+    @Override
+    public int getDefaultPropertyIndex() {
+        return this.defaultPropertyIndex;
+    }
+
+    void mergeBeanInfo(BeanInfo beanInfo, boolean force)
+            throws IntrospectionException {
+        if (force || !explicitProperties) {
+            PropertyDescriptor[] superDescs = beanInfo.getPropertyDescriptors();
+            if (superDescs != null) {
+                if (getPropertyDescriptors() != null) {
+                    properties = mergeProps(superDescs, beanInfo
+                            .getDefaultPropertyIndex());
+                } else {
+                    properties = superDescs;
+                    defaultPropertyIndex = beanInfo.getDefaultPropertyIndex();
+                }
+            }
+        }
+
+        if (force || !explicitMethods) {
+            MethodDescriptor[] superMethods = beanInfo.getMethodDescriptors();
+            if (superMethods != null) {
+                if (methods != null) {
+                    methods = mergeMethods(superMethods);
+                } else {
+                    methods = superMethods;
+                }
+            }
+        }
+
+        if (force || !explicitEvents) {
+            EventSetDescriptor[] superEvents = beanInfo
+                    .getEventSetDescriptors();
+            if (superEvents != null) {
+                if (events != null) {
+                    events = mergeEvents(superEvents, beanInfo
+                            .getDefaultEventIndex());
+                } else {
+                    events = superEvents;
+                    defaultEventIndex = beanInfo.getDefaultEventIndex();
+                }
+            }
+        }
+    }
+
+    /* MODIFIED FOR THE MSGPACK PROJECT
+     * merge the PropertyDescriptor with superclass
+     */
+    private PropertyDescriptor[] mergeProps(PropertyDescriptor[] superDescs,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 81. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 459-470 into a separate method.